VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -234,8 +243,8 @@ def post_to_wireserver(self, data):
                 else:
                     raise Exception("no response from encryption settings update request")
             except Exception as e:
+                retry_count += 1
                 if retry_count < retry_count_max:
-                    retry_count += 1
                     time.sleep(5) # sleep for 5 seconds before retrying.
                 else:
                     raise e
```

Comment:
To avoid no boot scenarios later, the encryption operation should fail at this point (if after multiple retries the encryption settings could not be uploaded to the host).  Rather than re-raising the exception and hoping the caller does this, maybe it would be better to simply exit with failure right at this point ensuring the properly telemetry and status is being reflected.

Reply:
post_to_wireserver is a method used at multiple places and it would be hard to error out here with the right error codes. So, I want to keep the onus on the caller to exit with the right code and operation string.

Reply:
I haven't had a chance to review every caller yet, but that may be necessary since it is unsafe to allow encryption to proceed if encryption settings haven't been stamped to disk.

Reply:
I looked at a the callers on this method.
And this does always halt the encryption process.
The error codes are "encryption_failed" or "unknown_error". Which while unfortunate.. is kind of true. When the "post_to_wireserver" fails, at this time, we actually do not know why is failed.